zebrad/tests/common/get_block_template_rpcs.rs

```diff
@@ -0,0 +1,164 @@
+use super::*;
+
+pub(crate) mod submit_block {
+
+    use std::path::PathBuf;
+
+    use color_eyre::eyre::{eyre, Context, Result};
+
+    use futures::TryFutureExt;
+    use indexmap::IndexSet;
+    use reqwest::Client;
+    use tower::{Service, ServiceExt};
+    use zebra_chain::{block::Height, parameters::Network, serialization::ZcashSerialize};
+    use zebra_state::HashOrHeight;
+    use zebra_test::args;
+
+    use crate::common::{
+        cached_state::{copy_state_directory, start_state_service_with_cache_dir},
+        config::{persistent_test_config, testdir},
+        launch::ZebradTestDirExt,
+        lightwalletd::random_known_rpc_port_config,
+    };
+
+    use super::cached_state::{load_tip_height_from_state_directory, ZEBRA_CACHED_STATE_DIR};
+
+    async fn get_future_block_hex_data(
```

Comment:
It looks like this code has been copied from the send transaction test, but some details have been missed. I'll add some examples below.

I'd like to rewrite this test using the shared RPC test framework. We might need to add a new `LightwalletdTestType` variant (and maybe rename that enum). But I think the end result will be more maintainable.

Reply:
I'll make these updates in #5015 when adding the CI job, please skip reviewing this draft of the acceptance test in detail until then.